3-extra/2-array-of-objects.js

```diff
@@ -10,73 +10,80 @@
     Each title in the resulting array should be the highest rated book in its genre.
 */
 
-function getHighestRatedInEachGenre(books) {
-    // TODO
+const getHighestRatedInEachGenre = (books) => {
```

Comment:
Good thinking, and this does work for the test case, but that is more so because the test isn't designed very well. 
What you're doing here is:

1. Sorting the books by rating
2. Store the title of the books, still in order by their rating
3. Print off the first 3

This is passing the test because after sorting, it just so happens that the three genres the test is using, show up in the top 3. If you increase the value of the Harry Potter book to 5, and re run it, you'll see you now have 2 children's genre books being returned which is not what this function is meant to do. 

This would also break if another genre is added as you are hardcoding how many to return (3), rather than counting for how many unique genres are in the set. 

Dont be afraid to add some logic in there to make it easier to follow - there's no requirement to make it all fit in embedded return statements (they are quite difficult to follow and find where errors are happening)

The approach I would take to this problem is to create a dictionary of genres, and in that store all the details about the highest rating book, in that genre. As we loop through the BOOKS, we will check the dictionary for a matching genre. If there is a matching genre, we compare to see if the currently stored book has a lower rating than the indexed book - if so, we'll store out new higher rated book in that genre. If there is no genre match, then we know, by virtue of no competition, that that is currently the highest rated book in that genre.

Once we are done looping, we can print off just the titles from within that dictionary.